internal/provider/resource_apply_module.go

```diff
@@ -0,0 +1,591 @@
+func NewApplyModuleResource() resource.Resource {
+	return &applyModuleResource{}
+}
+
+type applyModuleResource struct {
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *applyModuleResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_apply_module"
+}
+
+func (t *applyModuleResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	description := "Defines and manages an apply module."
+
+	resp.Schema = schema.Schema{
+		Version:             1,
+		MarkdownDescription: description,
+		Description:         description,
+		Attributes: map[string]schema.Attribute{
+			"workspace_path": schema.StringAttribute{
+				MarkdownDescription: "The full path of the workspace.",
+				Description:         "The full path of the workspace.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_source": schema.StringAttribute{
+				MarkdownDescription: "The source of the module.",
+				Description:         "The source of the module.",
+				Required:            true,
+			},
+			"module_version": schema.StringAttribute{
+				MarkdownDescription: "The version identifier of the module.",
+				Description:         "The version identifier of the module.",
+				Optional:            true,
+				Computed:            true, // computed if not supplied
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"variables": schema.ListNestedAttribute{
+				MarkdownDescription: "Optional list of variables for the run in the target workspace.",
+				Description:         "Optional list of variables for the run in the target workspace.",
+				Optional:            true,
+				Computed:            true, // Terraform requires it to be computed if it's optional.
+				PlanModifiers: []planmodifier.List{
+					modifiers.ListDefault([]attr.Value{}),
+				},
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"value": schema.StringAttribute{
+							MarkdownDescription: "Value of the variable.",
+							Description:         "Value of the variable.",
+							Required:            true,
+						},
+						"key": schema.StringAttribute{
+							MarkdownDescription: "Key or name of this variable.",
+							Description:         "Key or name of this variable.",
+							Required:            true,
+						},
+						"category": schema.StringAttribute{
+							MarkdownDescription: "Category of this variable, 'terraform' or 'environment'.",
+							Description:         "Category of this variable, 'terraform' or 'environment'.",
+							Required:            true,
+						},
+						"hcl": schema.BoolAttribute{
+							MarkdownDescription: "Whether this variable is HCL (vs. environment).",
```

Comment:
I believe this should always be true but I guess that depends on how a string is handled. Also the description needs updated as it has nothing to do with environment.

Reply:
Again, this is marked as resolved but I see no explanation as to why there has been no change.